src/index.ts

```diff
@@ -176,10 +201,169 @@ async function startServiceWorker(ses: Electron.Session, extension: Electron.Ext
   }
 }
 
+function patchIpcMain() {
+  const listenerMap = new Map<string, Map<any, any>>(); // channel -> (originalListener -> tracked/cleaned Listener)
```

Comment:
Appreciate the comment here!
nit: It might be nice to create named types to make this code more self documenting:
```suggestion
// top level
type Channel = string
type Listener = any
...
  const listenerMap = new Map<Channel, Map<Listener, Listener>>(); // channel -> (originalListener -> tracked/cleaned Listener)
```